changes/ocd-3663.sql

```diff
@@ -0,0 +1,117 @@
+
+CREATE TRIGGER criterion_listing_statistic_audit AFTER INSERT OR UPDATE OR DELETE on openchpl.criterion_listing_statistic FOR EACH ROW EXECUTE PROCEDURE audit.if_modified_func();
+CREATE TRIGGER criterion_listing_statistic_timestamp BEFORE UPDATE on openchpl.criterion_listing_statistic FOR EACH ROW EXECUTE PROCEDURE openchpl.update_last_modified_date_column();
+
+CREATE TABLE openchpl.criterion_upgraded_from_original_listing_statistic (
+	id bigserial NOT NULL,
+	listing_count bigint NOT NULL,
+	certification_criterion_id bigint NOT NULL,
+	statistic_date date NOT NULL, -- the date to which this statistic applies
+	creation_date timestamp without time zone NOT NULL DEFAULT now(),
+	last_modified_date timestamp without time zone NOT NULL DEFAULT now(),
+	last_modified_user bigint NOT NULL,
+	deleted boolean NOT NULL DEFAULT false,
+	CONSTRAINT criterion_upgraded_from_original_listing_statistic_pk PRIMARY KEY (id),
+	CONSTRAINT certification_criterion_fk FOREIGN KEY (certification_criterion_id)
+		REFERENCES openchpl.certification_criterion (certification_criterion_id)
+		MATCH FULL ON DELETE CASCADE ON UPDATE CASCADE
+);
+CREATE INDEX idx_criterion_upgraded_from_original_listing_stat_date on openchpl.criterion_upgraded_from_original_listing_statistic (statistic_date);
+
+CREATE TRIGGER criterion_upgraded_from_original_listing_statistic_audit AFTER INSERT OR UPDATE OR DELETE on openchpl.criterion_upgraded_from_original_listing_statistic FOR EACH ROW EXECUTE PROCEDURE audit.if_modified_func();
+CREATE TRIGGER criterion_upgraded_from_original_listing_statistic_timestamp BEFORE UPDATE on openchpl.criterion_upgraded_from_original_listing_statistic FOR EACH ROW EXECUTE PROCEDURE openchpl.update_last_modified_date_column();
+
+CREATE TABLE openchpl.cures_criterion_upgraded_without_original_listing_statistic (
+	id bigserial NOT NULL,
+	listing_count bigint NOT NULL,
+	certification_criterion_id bigint NOT NULL,
+	statistic_date date NOT NULL, -- the date to which this statistic applies
+	creation_date timestamp without time zone NOT NULL DEFAULT now(),
+	last_modified_date timestamp without time zone NOT NULL DEFAULT now(),
+	last_modified_user bigint NOT NULL,
+	deleted boolean NOT NULL DEFAULT false,
+	CONSTRAINT cures_criterion_upgraded_without_original_listing_statistic_pk PRIMARY KEY (id),
+	CONSTRAINT certification_criterion_fk FOREIGN KEY (certification_criterion_id)
+		REFERENCES openchpl.certification_criterion (certification_criterion_id)
+		MATCH FULL ON DELETE CASCADE ON UPDATE CASCADE
+);
+CREATE INDEX idx_cures_criterion_upgraded_without_original_listing_stat_date on openchpl.cures_criterion_upgraded_without_original_listing_statistic (statistic_date);
+
+CREATE TRIGGER cures_criterion_upgraded_without_original_listing_statistic_audit AFTER INSERT OR UPDATE OR DELETE on openchpl.cures_criterion_upgraded_without_original_listing_statistic FOR EACH ROW EXECUTE PROCEDURE audit.if_modified_func();
```

Comment:
I get this `NOTICE` on this line, and something similar on the one below:

```
psql:./changes/ocd-3663.sql:62: NOTICE:  identifier "cures_criterion_upgraded_without_original_listing_statistic_audit" will be truncated to "cures_criterion_upgraded_without_original_listing_statistic_aud"
```

Is this a problem?

Reply:
I don't think it's a *problem* in that the triggers are still created and will function. They just don't have the names we would expect because they are too long and get truncated. The table names were so long because I couldn't figure out how to make them describe what they were otherwise. I will make the trigger names shorter.